src/lib.rs

```diff
@@ -407,3 +408,63 @@ impl_single!(f64);
 
 impl_single!(usize);
 impl_single!(bool);
+
+impl<const N: usize> Miniconf for String<N> {
```

Comment:
This indeed looks identical to the implementation for `impl_single! with the caveat that it has the const-generic surrounding it. There's likely a way we can wrap these together. I'm happy to take over if you don't want to worry about it.

Reply:
Cool, I spent a little while looking into macros but I decided it's too long of an endeavor for me right now if you maybe already know how to.